actor/src/main/scala/org/apache/pekko/util/UniqueRandomShortProvider.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * license agreements; and to You under the Apache License, version 2.0:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * This file is part of the Apache Pekko project, derived from Akka.
+ */
+
+package org.apache.pekko.util
+
+import org.apache.pekko.annotation.InternalApi
+
+import java.security.SecureRandom
+
+/**
+ * An efficient collision free threadsafe generator of pseudo random
+ * shorts based on https://en.wikipedia.org/wiki/Double_hashing#Enhanced_double_hashing
+ * using SecureRandom as a seed.
+ *
+ * The avoidance of collisions is achieved by using a Short Array buffer
+ * that is prefilled with a dimension of 65536 (~ 131072 bytes in size per
+ * instance of [[UniqueRandomShortProvider]]). This means it will loop
+ * through all possible 65536 values before starting to re-generating
+ * the same numbers again.
+ */
+
+@InternalApi
+private[pekko] class UniqueRandomShortProvider {
+  import UniqueRandomShortProvider._
+
+  // These methods are volatile so that they stay within cache
+  @volatile private var index = 0L
```

Comment:
I think @Claudenw agreed that the volatiles should go

Reply:
So I just spoke with him and he said that he did put volatile there deliberately, its a trick to keep the cache warm (see https://github.com/apache/incubator-pekko/pull/371#discussion_r1228105897).

I will do another set of benchmarks without volatile to see if it makes a difference

Reply:
I removed them as they don't seem to be making a difference